packages/aws-cdk-lib/aws-stepfunctions/lib/states/distributed-map.ts

```
@@ -0,0 +1,770 @@
+          's3:ListMultipartUploadParts',
+          's3:AbortMultipartUpload',
+        ],
+        resources: [resource],
+      }),
+    ];
+  }
+}
+
+/**
+ * Interface for ItemBatcher configuration properties
+ */
+export interface ItemBatcherProps {
+  /**
+   * MaxItemsPerBatch
+   *
+   * Specifies the maximum number of items that each child workflow execution processes, as static number
+   *
+   * @default - No maxItemsPerBatch
+   */
+  readonly maxItemsPerBatch?: number;
+
+  /**
+   * MaxItemsPerBatchPath
+   *
+   * Specifies the maximum number of items that each child workflow execution processes, as JsonPath
+   *
+   * @default - No maxItemsPerBatchPath
+   */
+  readonly maxItemsPerBatchPath?: string;
+
+  /**
+   * MaxInputBytesPerBatch
+   *
+   * Specifies the maximum number of bytes that each child workflow execution processes, as static number
+   *
+   * @default - No maxInputBytesPerBatch
+   */
+  readonly maxInputBytesPerBatch?: number;
+
+  /**
+   * MaxInputBytesPerBatchPath
+   *
+   * Specifies the maximum number of bytes that each child workflow execution processes, as JsonPath
+   *
+   * @default - No maxInputBytesPerBatchPath
+   */
+  readonly maxInputBytesPerBatchPath?: string;
+
+  /**
+   * BatchInput
+   *
+   * Fixed JSON input to include in each batch passed to each child workflow execution
+   *
+   * @default - No batchInput
+   */
+  readonly batchInput?: object;
+}
+
+/**
+ * Configuration for processing a group of items in a single child workflow execution
+ */
+export class ItemBatcher {
+  private props: ItemBatcherProps;
+
+  constructor(props: ItemBatcherProps) {
+    this.props = props;
+  }
+
+  /**
+   * Render ResultWriter in ASL JSON format
+   */
+  public render(): any {
+    if (
+      !this.props.maxItemsPerBatch &&
+      !this.props.maxItemsPerBatchPath &&
+      !this.props.maxInputBytesPerBatch &&
+      !this.props.maxInputBytesPerBatchPath &&
+      !this.props.batchInput) {
+      return undefined;
```

We should throw an error here if nothing is specified right? Why is this not a condition of the `validate` function

---

Yes, I'll change that up